### PR TITLE
Realized LinkedList class with Node class and basic functions. Also c…

### PR DESCRIPTION
**Pull request by Trayvor**
…reated 2 private functions: getIndexedElement() that helps to find element by index and unlink() that actually deletes item.

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -3,45 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
```
Comment (okuzan) [inline src/main/java/core/basesyntax/MyLinkedList.java:8]:
```suggestion
    private int size;
```
0 is default for int

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -3,45 +3,143 @@
...
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            return super.equals(obj);
+        }
```
Comment (okuzan) [inline src/main/java/core/basesyntax/MyLinkedList.java:143]: we don't need equals here, and what's the point to override it if you leave the same implementation

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 3]
```diff
@@ -3,45 +3,143 @@
...
+        return deletedNode.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> deletedNode;
+        for (int i = 0; i < size; i++) {
+            deletedNode = getIndexedElement(i);
+            if ((deletedNode.item != null && deletedNode.item.equals(object))
+                    || (deletedNode.item == null && object == null)) {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/MyLinkedList.java:72]: extra parentheses

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 4]
```diff
@@ -3,45 +3,143 @@
...
+            return;
+        }
+        newElement.prev.next = newElement;
+        indexedElement.prev = newElement;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/MyLinkedList.java:42]:
```suggestion
        for (T note : list) {
```

**Review by okuzan — CHANGES_REQUESTED** [review]
Good job overall!

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 2 above)
Comment (Trayvor) [reply src/main/java/core/basesyntax/MyLinkedList.java:143]: oops i forgot to delete that

**Review by olekskov — APPROVED** [review]
Good job

**Pull request closed without merging**